lib/roo_on_rails/checks/base.rb

```diff
@@ -0,0 +1,72 @@
+require 'roo_on_rails/checks'
+require 'roo_on_rails/checks/helpers'
+
+module RooOnRails
+  module Checks
+    class Base
+      include Helpers
+
+      def self.run(**options)
+        new(**options).run
+      end
+
+      def initialize(fix: false, state: nil)
+        @fix = fix
+        @state = state
+      end
+
+      def run
+        say _intro
+        _call
+      rescue Failure
+        raise unless @fix
+        say "\t· attempting to fix", %i[yellow]
+        _fix
+        say "\t· re-checking", %i[yellow]
+        _call
+      end
+
+      protected
+
+      def _intro
```

Comment:
As you know I'm generally not keen on the leading underscore in Ruby as it doesn't seem to be a common convention. I can just about live with it for private methods, but it seems egregious when used for methods that are _intended_ to be overridden by other classes.